File: contest/牛客网PAT模拟题/BasicLevel_Real/1002.py
List = list(map(int, input().strip().split()))
flag = False
Is = [False, False, False, False, False]
ans = [0, 0, 0, [], 0]
for i in range(1, List[0] + 1):
    if (List[i] % 10 < 1):
        Is[0] = True
        ans[0] += List[i]
    if (List[i] % 5 == 1):
        Is[1] = True
        ans[1] += (-1)**int(flag) * List[i]
        flag = not flag
    if (List[i] % 5 == 2):
        Is[2] = True
        ans[2] += 1
    if (List[i] % 5 == 3):
        Is[3]= True
        ans[3].append(List[i])
    if (List[i] % 5 == 4):
        Is[4] = True
        ans[4]= max(ans[4], List[i])

print(ans[0] if Is[0] else 'N', ans[1] if Is[1] else 'N', ans[2] if Is[2] else 'N',
      ('%.1f' % (sum(ans[3])/len(ans[3]))) if Is[0] else 'N', ans[4] if Is[4] else 'N')

File: contest/牛客网PAT模拟题/BasicLevel_Real/1002.cpp
#include <algorithm>
#include <iostream>
using namespace std;
int list[10000];
int main()
{
    int n, a[5] = {0}, cnt = 0;
    scanf("%d", &n);
    bool flag[6] = {false};
    for (int i = 0; i < n; i++)
        scanf("%d", list + i);
    for (int i = 0; i < n; i++)
    {
        if (!(list[i] % 10))
        {
            flag[0] = true;
            a[0] += list[i];
        }
        else if (list[i] % 5 == 1)
        {
            flag[1] = true;
            if (flag[5])
                a[1] -= list[i];
            else
                a[1] += list[i];
            flag[5] = !flag[5];
        }
        else if (list[i] % 5 == 2)
        {
            flag[2] = true;
            a[2]++;
        }
        else if (list[i] % 5 == 3)
        {
            flag[3] = true;
            a[3] += list[i];
            cnt++;
        }
        else if (list[i] % 5 == 4)
        {
            flag[4] = true;
            a[4] = max(a[4], list[i]);
        }
    }
    if (flag[0])
        printf("%d ", a[0]);
    else
        printf("N ");
    if (flag[1])
        printf("%d ", a[1]);
    else
        printf("N ");
    if (flag[2])
        printf("%d ", a[2]);
    else
        printf("N ");
    if (flag[3])
        printf("%.1f ", double(a[3]) / cnt);
    else
        printf("N ");
    if (flag[4])
        printf("%d\n", a[4]);
    else
        printf("N\n");
    return 0;
}
